test/feature-benchmark/scenarios.py

```diff
@@ -572,6 +572,52 @@ def benchmark(self) -> MeasurementSource:
         )
 
 
+class AccumulateReductions(Dataflow):
+    """Benchmark the accumulation of reductions."""
+
+    def before(self) -> Action:
+        return TdAction(
+            """
+> DROP TABLE IF EXISTS t CASCADE;
+> CREATE TABLE t (a int, b int, c int, d int);
+
+> CREATE MATERIALIZED VIEW data AS
+  SELECT a, a AS b FROM generate_series(1, 10000000) AS a
+  UNION ALL
+  SELECT a, b FROM t;
+
+> INSERT INTO t (a, b) VALUES (1, 1);
+> INSERT INTO t (a, b) VALUES (0, 0);
+
+> DROP CLUSTER IF EXISTS idx_cluster CASCADE;
+> CREATE CLUSTER idx_cluster SIZE '1-8G', REPLICATION FACTOR 1;
+
+> CREATE VIEW accumulable AS
+  SELECT
+    a,
+    sum(a) AS sum_a, count(a) as cnt_a,
+    sum(b) AS sum_b, count(b) as cnt_b
+  FROM data
+  GROUP BY a;
+"""
+        )
+
+    def benchmark(self) -> MeasurementSource:
+        return Td(
+            """
+> SELECT 1
+  /* A */
+1
+
+> CREATE INDEX i_accumulable IN CLUSTER idx_cluster ON accumulable(a);
```

Comment:
I was wondering if this should be part of the benchmark or not but it does not seem to have a big influence on the results.

Not included in benchmark:
```
AccumulateReductions                | wallclock |          13.791 |          14.696 |      no       | 6.2 pct   less/faster
AccumulateReductions                | messages  |   110000617.000 |   110000719.000 |      no       | 0.0 pct   less/faster
AccumulateReductions                | memory    |        8190.155 |        7944.107 |      no       | 3.1 pct   more/slower
```

Included in benchmark:
```
AccumulateReductions                | wallclock |          13.641 |          14.260 |      no       | 4.3 pct   less/faster
AccumulateReductions                | messages  |   110000668.000 |   110000617.000 |      no       | 0.0 pct   more/slower
AccumulateReductions                | memory    |        8259.773 |        8039.474 |      no       | 2.7 pct   more/slower
```

Reply:
The pattern is to include the `CREATE INDEX` in the benchmarked region. This way one can be certain that the entire time the dataflow is hydrating is captured in the `wallclock`. Otherwise it would have been possible that by the time `CREATE INDEX` returns, a lot of hydration work would have completed asynchronously.